pkg/disttask/framework/scheduler/scheduler_manager.go

```diff
@@ -394,6 +397,7 @@ func (sm *Manager) cleanupFinishedTasks(tasks []*proto.Task) error {
 		failpoint.Return(errors.New("transfer err"))
 	})
 
+	sm.collectMetrics()
```

Comment:
```suggestion
```
our metric is already not real-time value, i don't think we need this trigger, it couples them

Reply:
`GetAllSubtasks` gets the subtask from `tidb_background_subtask`. After `TransferTasks2History`, we cannot obtain the previously recorded subtasks, so I choose to clean up the previously recorded subtasks before `TransferTasks2History`.

Reply:
in our scenario, a self-defined metric collector wound be better, as some subtask-id/exec-id combination might not exist any more, so we can generate metric with labels on the fly